File: Neet/Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        res = 0

        for n in nums:
            if n - 1 not in nums:
                count = 1
                while n + 1 in nums:
                    count += 1
                    n += 1
                res = max(res, count)

        return res
        

File: Leetcode/topQuestions/LongestConsecutiveSequence.java
package Leetcode.topQuestions;

import java.util.HashSet;
import java.util.Set;

public class LongestConsecutiveSequence {
    public int longestConsecutive(int[] nums) {

        Set<Integer> numSet = new HashSet<>();
        int res = 0;

        for(int num: nums){
            numSet.add(num);
        }

        for(int num: numSet){
            if(!numSet.contains(num - 1)){
                int count = 1;
                while(numSet.contains(num + 1)){
                    count++;
                    num++;
                }

                res= Math.max(count, res)
            }
        }

        return res;
    }

}